hoot-core/src/main/cpp/hoot/core/util/Boundable.h

```diff
@@ -49,7 +57,7 @@ class Boundable
    * Sets the bounds in WGS84. It is important to note that even if the rest of the operation is
    * not in WGS84 the bounds will still be in WGS84.
    */
-  virtual void setBounds(const geos::geom::Envelope& bounds) = 0;
+  virtual void setBounds(const std::shared_ptr<geos::geom::Geometry>& bounds) = 0;
 };
```

Comment:
The main change of the whole PR.

Reply:
It might be too late for this but I saw all of the places where a call to `setBounds(bounds)` became `setBounds(GeometryUtils::envelopeToPolygon(bounds))`.  I think it would have been better just to override `setBounds` instead of replace it and convert the envelope in one place.  Something like this:
```
  void setBounds(const geos::geom::Envelope& bounds)
  { setBounds(GeometryUtils::envelopeToPolygon(bounds)); }
  virtual void setBounds(const std::shared_ptr<geos::geom::Geometry>& bounds) = 0;
```
Then all of the derived classes would only have to implement the geometry version and any class that calls it with an Envelope automatically gets the conversion.  It might make for a little cleaner code.  Just an idea.

Reply:
That's a good idea. I'll do that.